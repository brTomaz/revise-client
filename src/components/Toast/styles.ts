import styled, { css, DefaultTheme } from 'styled-components'
import { ToastContainer } from 'react-toastify'

export const BaseToast = (theme: DefaultTheme) => css`
  background-color: ${theme.colors.primary};
  border-radius: 0.5rem;
  color: ${theme.colors.white};

  div {
    display: flex;
    flex-direction: column;
    align-items: center;

    svg {
      width: 4rem;
    }
  }
`

export const Wrapper = styled(ToastContainer)`
  .Toastify__toast--success {
    ${({ theme }) => css`
      ${BaseToast(theme)};
      svg {
        color: ${theme.colors.white};
      }
    `}
  }
`

export const Points = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    border-radius: 7px;
    font-size: ${theme.font.sizes.xhuge};
    font-weight: ${theme.font.bold};
    text-align: center;
    color: ${theme.colors.primary};
    padding: ${theme.spacings.xsmall};
    margin: ${theme.spacings.xsmall} 0;
  `}
`
export const Attempt = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.bold};
  `}
`
