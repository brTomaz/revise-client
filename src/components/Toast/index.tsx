import { toast } from 'react-toastify'
import Lottie from 'react-lottie'

import animation from 'assets/correct-animation.json'
import * as Styles from './styles'

export type ToastProps = {
  points: number
  attempt: number
}

const defaultOptions = {
  loop: false,
  animationData: animation
}

export const showToast = ({ points, attempt }: ToastProps) => {
  toast.success(
    <div>
      <Lottie options={defaultOptions} height={96} width={96} />
      <Styles.Points>+{points}</Styles.Points>
      <Styles.Attempt>Acertou na {attempt}Âª tentativa.</Styles.Attempt>
    </div>
  )
}

const Toast = () => (
  <Styles.Wrapper
    hideProgressBar
    limit={1}
    autoClose={3000}
    pauseOnHover={false}
    pauseOnFocusLoss={false}
  />
)

export default Toast
