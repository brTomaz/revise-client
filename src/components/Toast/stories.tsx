import { Story, Meta } from '@storybook/react/types-6-0'
import Toast, { showToast } from '.'

export default {
  title: 'Toast',
  component: Toast
} as Meta

showToast({ points: 5, attempt: 1 })
export const Default: Story = () => <Toast />
