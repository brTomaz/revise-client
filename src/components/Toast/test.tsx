import { renderWithTheme } from 'utils/tests/helpers'
import Toast from '.'

describe('<Toast />', () => {
  it('should Toast render with styles', () => {
    const { container } = renderWithTheme(<Toast />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
