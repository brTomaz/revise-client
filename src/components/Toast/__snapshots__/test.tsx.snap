// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> should Toast render with styles 1`] = `
@media (min-width:768px) {

}

<div
  class="Toastify"
/>
`
